src/hooks/useDrag.tsx

```diff
@@ -0,0 +1,418 @@
+        draggedError
+      )
+
+      const displayBool = displaySections[source.index]
+      const newDisplaySections = updatePositions(
+        displaySections,
+        source.index,
+        destination.index,
+        displayBool
+      )
+
+      return updateEditorSection(
+        homepageState,
+        newDisplaySections,
+        newSections,
+        newSectionErrors
+      )
+    }
+    // inner dnd for hero
+    // passed in via droppableId
+    case "dropdownelem": {
+      // TODO: type check to avoid casting
+      const draggedElem = (frontMatter.sections[0]
+        .hero as EditorHeroDropdownSection).dropdown.options[source.index]
+      //! Thsis needs to be updated
+      const newDropdownOptions = updatePositions(
+        (frontMatter.sections[0].hero as EditorHeroDropdownSection).dropdown
+          .options,
+        source.index,
+        destination.index,
+        draggedElem
+      )
+
+      const draggedError = errors.dropdownElems[source.index]
+      const newDropdownErrors = updatePositions(
+        errors.dropdownElems,
+        source.index,
+        destination.index,
+        draggedError
+      )
+      const displayBool = displayDropdownElems[source.index]
+      const newDisplayDropdownElems = updatePositions(
+        displayDropdownElems,
+        source.index,
+        destination.index,
+        displayBool
+      )
+
+      return {
+        ...homepageState,
+        displayDropdownElems: newDisplayDropdownElems,
+        frontMatter: {
+          ...frontMatter,
+          sections: _.set(
+            frontMatter.sections,
+            ["0", "hero", "dropdown", "options"],
+            newDropdownOptions
+          ),
+        },
+        errors: { ...errors, dropdownElems: newDropdownErrors },
+      }
+    }
+    case "highlight": {
+      // TODO: type check to avoid casting
+      const draggedElem = (frontMatter.sections[0]
+        .hero as EditorHeroHighlightsSection).key_highlights[source.index]
+      const newHighlightOptions = updatePositions(
+        (frontMatter.sections[0].hero as EditorHeroHighlightsSection)
+          .key_highlights,
+        source.index,
+        destination.index,
+        draggedElem
+      )
+
+      const draggedError = errors.highlights[source.index]
+      const newHighlightErrors = updatePositions(
+        errors.highlights,
+        source.index,
+        destination.index,
+        draggedError
+      )
+
+      const displayBool = displayHighlights[source.index]
+      const newDisplayHighlights = updatePositions(
+        displayHighlights,
+        source.index,
+        destination.index,
+        displayBool
+      )
+      return updateHighlightsSection(
+        homepageState,
+        newDisplayHighlights,
+        newHighlightOptions,
+        newHighlightErrors
+      )
+    }
+    default:
+      return homepageState
+  }
+}
+
+// NOTE: Handles only placement,
+// let the caller decide what to create
+export const onCreate = <E,>(
```

Comment:
quick qn, why do we type this to `E` when we are not referencing the type here ah

Reply:
we are referencing the type in the error isn't it

Reply:
we are referencing the type in the error isn't it